registration_api.js

```diff
@@ -0,0 +1,694 @@
+                        </td>
+                        `;
+        app_list_body.appendChild(tr);
+    });
+}
+
+const removeApp = async (app_id) => {
+    const endpoint = getEndpoint();
+    const api = new DerivAPIBasic({ endpoint, lang: 'EN', app_id });
+    const token1 = getToken();
+    await api.authorize(token1);
+    await api.appDelete(app_id);
+    send({ type: 'FETCH_APP_LIST' });
+}
+
+const appUpdate = async ({ app_id, name, redirect_uri, scopes }) => {
+    const endpoint = getEndpoint();
+    const api = new DerivAPIBasic({ endpoint, lang: 'EN', app_id });
+    const token1 = getToken();
+    await api.authorize(token1);
+    await api.appUpdate({ app_update: app_id, name, redirect_uri, scopes });
+    send({ type: 'FETCH_APP_LIST' });
+};
+
+const open_delete_dialog = (app_id) => {
+    const dialog = document.getElementById('delete_app_dialog');
+    dialog.showModal();
+    const delete_app_button = document.getElementById('delete_app_button');
+    delete_app_button.addEventListener('click', () => {
+        send({ type: "DELETE_APP", data: app_id });
+        dialog.close();
+    });
+}
+
+const open_update_dialog = (app_id, name, scopes, redirect_uri) => {
+    const dialog = document.getElementById('update_app_dialog');
+    dialog.showModal();
+    const update_name = document.getElementById('update_app_name');
+    update_name.value = name;
+    const update_redirect_uri = document.getElementById('update_app_redirect_uri');
+    update_redirect_uri.value = redirect_uri;
+    const update_app_button = document.getElementById('update_app_button');
+
+    const update_read_scope = document.getElementById('update_read-scope');
+    if (scopes.includes('read')) {
+        update_read_scope.checked = true;
+    }
+    const update_trade_scope = document.getElementById('update_trade-scope');
+    if (scopes.includes('trade')) {
+        update_trade_scope.checked = true;
+    }
+    const update_trading_information_scope = document.getElementById('update_trading_information-scope');
+    if (scopes.includes('trading_information')) {
+        update_trading_information_scope.checked = true;
+    }
+    const update_payments_scope = document.getElementById('update_payments-scope');
+    if (scopes.includes('payments')) {
+        update_payments_scope.checked = true;
+    }
+    const update_admin_scope = document.getElementById('update_admin-scope');
+    if (scopes.includes('admin')) {
+        update_admin_scope.checked = true;
+    }
+
+    const checkedScopes = () => {
+        const checked_scopes = [];
+        const checkboxes = document.querySelectorAll('#app_scopes input[type="checkbox"]');
+        checkboxes.forEach((checkbox) => {
+            if (checkbox.checked) {
+                checked_scopes.push(checkbox.value);
+            }
+        });
+        return checked_scopes;
+    }
```

Comment:
I think this function can be more generic by accepting an id since there seems to be near similar logic at Line 615 below